app/models/reports/coda_finance_report.rb

```diff
@@ -0,0 +1,77 @@
+require 'csv'
+
+module Reports
+  # Used to generate reports summarising submissions in the format needed for
+  # feeding into Coda, the CCS finance system.
+  class CodaFinanceReport
+    attr_reader :submissions
+
+    HEADERS = [
+      'RunID',
+      'Nominal',
+      'Customer Code',
+      'Customer Name',
+      'Contract ID',
+      'Order Number',
+      'Lot Description',
+      'Inf Sales',
+      'Commission',
+      'Commission %',
+      'End User',
+      'Submitter',
+      'Month',
+      'M_Q'
+    ].freeze
+
+    def initialize(submissions)
+      @submissions = submissions
+    end
+
+    def to_csv
+      CSV.generate write_headers: true, headers: HEADERS do |csv|
+        submissions.each do |submission|
+          csv << row_for_submission(submission, Customer.sectors[:central_government])
+          csv << row_for_submission(submission, Customer.sectors[:wider_public_sector])
+        end
+      end
+    end
+
+    private
+
+    def row_for_submission(submission, sector)
+      data = CodaFinanceReportRow.new(submission, sector).data
+
+      [
+        report_uid,
+        data['Nominal'],
+        data['Customer Code'],
+        data['Customer Name'],
+        data['Contract ID'],
+        data['Order Number'],
+        data['Lot Description'],
+        data['Inf Sales'],
+        data['Commission'],
+        data['Commission %'],
+        data['End User'],
+        data['Submitter'],
+        data['Month'],
+        monthly_or_quarterly
+      ]
+    end
+
+    # CCS require that each report has a unique reference in case they need to
+    # perform an audit and trace a particular entry in the finance system back
+    # to the report that generated it.
+    def report_uid
+      "DSS-#{Time.zone.now.to_i}"
```

Comment:
Should the prefix here be "RMI", now we're no longer calling the service "Data Submission Service" ?

Reply:
Good suggestion, I'll update.